src/main/kotlin/no/nav/klage/oppgave/service/distribusjon/KlagebehandlingDistribusjonService.kt

```diff
@@ -0,0 +1,119 @@
+package no.nav.klage.oppgave.service.distribusjon
+
+import no.nav.klage.oppgave.domain.klage.BrevMottaker
+import no.nav.klage.oppgave.domain.klage.Klagebehandling
+import no.nav.klage.oppgave.domain.klage.Vedtak
+import no.nav.klage.oppgave.service.KlagebehandlingService
+import no.nav.klage.oppgave.util.getLogger
+import no.nav.klage.oppgave.util.getSecureLogger
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Propagation
+import org.springframework.transaction.annotation.Transactional
+import java.util.*
+
+@Service
+@Transactional
+class KlagebehandlingDistribusjonService(
+    private val klagebehandlingService: KlagebehandlingService,
+    private val vedtakDistribusjonService: VedtakDistribusjonService,
+    private val klagebehandlingAvslutningService: KlagebehandlingAvslutningService
+) {
+
+    companion object {
+        @Suppress("JAVA_CLASS_ON_COMPANION")
+        private val logger = getLogger(javaClass.enclosingClass)
+        private val secureLogger = getSecureLogger()
+    }
+
+    @Transactional(propagation = Propagation.NEVER)
```

Comment:
Hvorfor `NEVER`?

Reply:
For å sikre at det ikke er noen transaksjons-context som de andre under-transaksjonene blir med i, da ville det blitt en stor transaksjon, og det ønsker vi ikke.